src/internal/pfsdb/common.go

```diff
@@ -75,13 +82,18 @@ func (i *pageIterator[T]) hasNext() bool {
 	return i.pageIdx < len(i.page)
 }
 
-func (i *pageIterator[T]) next(ctx context.Context, tx *pachsql.Tx) (*T, error) {
+func (i *pageIterator[T]) next(ctx context.Context, tx *pachsql.Tx) (*T, int64, error) {
```

Comment:
This is private so it's not an issue, but this doesn't implement `stream.Iterator` anymore which means none of the utilities in `stream` work.

Reply:
True, that's why I designed this as an internal private component. The intent of this private page iterator is just to serve as a helper function for the actual iterators that do implement `stream.Iterator`.

Ideally, this is fully generic so that anyone can build an iterator on top of this and iterate through our tables. However, I need to think about the design a bit more to satisfy all our custom behaviour like listing things with revision numbers.